Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/conceptcollection/ConceptCollectionDAO.java

```diff
@@ -42,376 +39,347 @@
-				List<ConceptsDTO> conceptsList = query2.list();
-				if(conceptsList != null && conceptsList.size() > 0)
-				{
-					sessionFactory.getCurrentSession().delete(conceptsList.get(0));
-				}
-			}
-			
-		}
-		catch(Exception e)
-		{
-			logger.info("deleteItems method :"+e.getMessage());	
-			throw new QuadrigaStorageException(e);
-		}
-		return errMsg;
-	}
-
-	/**
-	 * This method updates the concepts associated with the concept collection
-	 * @param : concept - concept associated with the concept collection
-	 * @param : collectionid - concept collection id
-	 * @param : username - logged in user
-	 * @return : String
-	 * @throws : QuadrigaStorageException
-	 */
-
-	public String updateItem(IConcept concept, String collectionId,String username) throws QuadrigaStorageException {
-		String errMsg = null;
-		if(concept == null || collectionId==null || username==null ||  collectionId.isEmpty() || username.isEmpty())
-		{
-			logger.error("updateItem: input argument IConceptCollection is null");
-			return null;
-		}
-		try
-		{
-			Query query = sessionFactory.getCurrentSession().createQuery("from ConceptsDTO c where c.item =:id");
-			query.setParameter("id", concept.getConceptId());
-			@SuppressWarnings("unchecked")
-			List<ConceptsDTO> conceptsDTOList = query.list();
-			
-			if(conceptsDTOList != null && conceptsDTOList.size() > 0)
-			{
-				ConceptsDTO conceptsDTO = conceptsDTOList.get(0);
-				conceptsDTO.setLemma(concept.getLemma());
-				conceptsDTO.setPos(concept.getPos());
-				conceptsDTO.setDescription(concept.getDescription());
-				conceptsDTO.setUpdateddate(new Date());
-				sessionFactory.getCurrentSession().update(conceptsDTO);
-			}
-			else
-			{
-				errMsg = "concept id is invalid.";
-			}
-		}
-		catch(Exception e)
-		{
-			logger.info("deleteItems method :"+e.getMessage());	
-			throw new QuadrigaStorageException(e);
-		}
-		return errMsg;
-	}
-
-	/**
-	 * This method retrieves the concept collection id associated with the given
-	 * concept collection name
-	 * @param : ccName - concept collection name
-	 * @return : String - concept collection id
-	 * @throws : QuadrigaStorageException
-	 */
-	@SuppressWarnings("unchecked")
-	@Override
-	public String getConceptCollectionId(String ccName) throws QuadrigaStorageException {
-		String ccID = null;
-		try
-		{
-			Query query = sessionFactory.getCurrentSession().getNamedQuery("ConceptCollectionDTO.findByCollectionname");
-			query.setParameter("collectionname", ccName);
-			List<ConceptCollectionDTO> ccIDList = query.list();
-			
-			if(ccIDList != null  && ccIDList.size()> 0)
-			{
-				ccID = ccIDList.get(0).getConceptCollectionid();
-			}
-		}
-		catch(Exception e)
-		{
-			logger.info("deleteItems method :"+e.getMessage());	
-			throw new QuadrigaStorageException(e);
-		}
-		return ccID;
-	}
-
-	
-	@Override
+                    }
+                }
+                collection.setConceptCollectionConcepts(concepts);
+            }
+        } catch (Exception e) {

```

Comment:
I know you didn't write the code, but let's improve it :)

- First of all, this should be the actual exception being thrown, but just exception. 
- Then, let's catch it where it's thrown. It should suffice to move this catch block to line 148.
- and last, remove the logger output in the next line.